### PR TITLE
Add Lda

### PR DESCRIPTION
**Pull request by joyeshmishra**
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/blbarker%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11275078%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23issuecomment-233777890%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222016-07-19T21%3A52%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/915825%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/joyeshmishra%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%3Agun%3A%22%2C%20%22created_at%22%3A%20%222016-07-19T21%3A52%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/915825%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/joyeshmishra%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222016-07-19T21%3A52%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11275078%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/blbarker%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-19T21%3A59%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11275078%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/blbarker%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-19T22%3A00%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11275078%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/blbarker%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222016-07-19T22%3A01%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11275078%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/blbarker%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%208b93dcff07120b15272d7c2ee761d3297c23032c%20core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71475028%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22-%20%5Bx%5D%20test1%5Cr%5Cn%5Cr%5Cn-%20%5Bx%5D%20test2%22%2C%20%22created_at%22%3A%20%222016-07-20T07%3A09%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/915825%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/joyeshmishra%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-07-20T15%3A53%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/915825%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/joyeshmishra%22%7D%7D%2C%20%7B%22body%22%3A%20%22test%22%2C%20%22created_at%22%3A%20%222016-07-26T17%3A17%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9339394%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/fmagnott%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala%3AL1-500%22%7D%2C%20%22Pull%207ddf746429a84f279240b3aa7833ec2df4a74044%20core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala%2022%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r72295255%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22test%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222016-07-26T17%3A17%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9339394%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/fmagnott%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala%3AL1-500%22%7D%2C%20%22Pull%208b93dcff07120b15272d7c2ee761d3297c23032c%20core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala%2024%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71430664%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22word%22%2C%20%22created_at%22%3A%20%222016-07-19T21%3A49%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11275078%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/blbarker%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala%3AL1-500%22%7D%2C%20%22Pull%208b93dcff07120b15272d7c2ee761d3297c23032c%20core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala%2026%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71432018%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%21fix%20important%22%2C%20%22created_at%22%3A%20%222016-07-19T21%3A57%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/915825%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/joyeshmishra%22%7D%7D%2C%20%7B%22body%22%3A%20%22%21fixed%22%2C%20%22created_at%22%3A%20%222016-07-19T21%3A59%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/915825%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/joyeshmishra%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-19T21%3A59%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11275078%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/blbarker%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-19T22%3A00%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11275078%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/blbarker%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-07-20T15%3A52%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/915825%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/joyeshmishra%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala%3AL1-500%22%7D%2C%20%22Pull%208b93dcff07120b15272d7c2ee761d3297c23032c%20core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala%2056%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r72185820%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22-%20%5B%20%5D%20Testing%20markdowns%22%2C%20%22created_at%22%3A%20%222016-07-26T04%3A20%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/915825%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/joyeshmishra%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala%3AL1-500%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23issuecomment-233777890%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23issuecomment-233777945%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23issuecomment-233778054%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23issuecomment-233779661%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23issuecomment-233780000%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23issuecomment-233780070%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71430664%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71432018%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71432206%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71432230%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71432313%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71475028%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71552647%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r71552748%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r72185820%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r72295162%22%2C%20%22https%3A//github.com/trustedanalytics/spark-tk/pull/61%23discussion_r72295255%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/blbarker'></a>
- [ ] <a href='#crh-comment-Pull 8b93dcff07120b15272d7c2ee761d3297c23032c core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala 24'></a> &nbsp;<b><a href='https://github.com/trustedanalytics/spark-tk/pull/61#discussion_r71430664'>File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:L1-500</a></b>
- <a href='https://github.com/blbarker'></a> word
- [ ] <a href='#crh-comment-Pull 8b93dcff07120b15272d7c2ee761d3297c23032c core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala 1'></a> &nbsp;<b><a href='https://github.com/trustedanalytics/spark-tk/pull/61#discussion_r71475028'>File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:L1-500</a></b>
- <a href='https://github.com/joyeshmishra'></a> - [x] test1
- [x] test2
- <a href='https://github.com/joyeshmishra'></a> :+1:
- <a href='https://github.com/fmagnott'></a> test
- [ ] <a href='#crh-comment-Pull 8b93dcff07120b15272d7c2ee761d3297c23032c core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala 56'></a> &nbsp;<b><a href='https://github.com/trustedanalytics/spark-tk/pull/61#discussion_r72185820'>File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:L1-500</a></b>
- <a href='https://github.com/joyeshmishra'></a> - [ ] Testing markdowns
- [ ] <a href='#crh-comment-Pull 7ddf746429a84f279240b3aa7833ec2df4a74044 core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala 22'></a> &nbsp;<b><a href='https://github.com/trustedanalytics/spark-tk/pull/61#discussion_r72295255'>File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:L1-500</a></b>
- <a href='https://github.com/fmagnott'></a> test
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/trustedanalytics/spark-tk/pull/61#issuecomment-233777890'>General Comment</a></b>
- <a href='https://github.com/joyeshmishra'></a> :shipit:
- <a href='https://github.com/joyeshmishra'></a> :shipit::gun:
- <a href='https://github.com/blbarker'></a> :shipit:
- <a href='https://github.com/blbarker'></a> :-1:<a href='#crh-update-none'></a>
- [x] <a href='#crh-comment-Pull 8b93dcff07120b15272d7c2ee761d3297c23032c core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala 26'></a> &nbsp;<b><a href='https://github.com/trustedanalytics/spark-tk/pull/61#discussion_r71432018'>File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:L1-500</a></b>
- <a href='https://github.com/joyeshmishra'></a> !fix important
- <a href='https://github.com/joyeshmishra'></a> !fixed
- <a href='https://github.com/blbarker'></a> :+1:<a href='#crh-update-complete'></a>
- <a href='https://github.com/blbarker'></a> :-1:<a href='#crh-update-complete'></a>

<a href='https://www.codereviewhub.com/trustedanalytics/spark-tk/pull/61?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/trustedanalytics/spark-tk/pull/61?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/trustedanalytics/spark-tk/pull/61?approve=0'></a>&nbsp;<a href='https://github.com/trustedanalytics/spark-tk/pull/61'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+import scala.collection.immutable.Map
+import scala.util.Try
+
+/**
+ * Model for Latent Dirichlet Allocation. This holds the Spark's DistributedLdaModel and other artifacts obtained
+ * during model training
+ *
+ * @param numTopics Number of topics in trained model
+ * @param documentColumnName Name of document column
+ * @param wordColumnName Name of word column
```
Comment (blbarker) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:24]: word

Comment (joyeshmishra) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala]: :+1:

**Comment by joyeshmishra** [comment]
:shipit:

**Comment by joyeshmishra** [comment]
:shipit::gun:

**Comment by blbarker** [comment]
:shipit:

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+ */
+case class TkLdaModel(numTopics: Int, documentColumnName: String, wordColumnName: String) {
```
Comment (joyeshmishra) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:26]: !fix important

Comment (joyeshmishra) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:26]: !fixed

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala (hunk 1 above)
Comment (blbarker) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala]: :-1:<a href='#crh-update-none'></a>

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala (hunk 2 above)
Comment (blbarker) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:26]: :+1:<a href='#crh-update-complete'></a>

**Comment by blbarker** [comment]
:-1:<a href='#crh-update-none'></a>

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala (hunk 1 above)
Comment (blbarker) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala]: :+1:<a href='#crh-update-none'></a>

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala (hunk 2 above)
Comment (blbarker) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:26]: :-1:<a href='#crh-update-complete'></a>

**Comment by blbarker** [comment]
:+1:<a href='#crh-update-none'></a>

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala (hunk 1 above)
Comment (blbarker) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala]: :-1:<a href='#crh-update-complete'></a>

**Comment by blbarker** [comment]
:shipit:

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala [hunk 3]
```diff
@@ -0,0 +1,499 @@
+package org.apache.spark.mllib.clustering.org.trustedanalytics.sparktk
```
Comment (joyeshmishra) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:1]: - [x] test1
- [x] test2

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala (hunk 2 above)
Comment (joyeshmishra) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:26]: :+1:

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala (hunk 3 above)
Comment (joyeshmishra) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala:1]: :+1:

**Comment by joyeshmishra** [comment]
:shipit:

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+   */
+  def this(distLdaModel: DistributedLDAModel, documentColumnName: String, wordColumnName: String) = {
+    this(distLdaModel.k, documentColumnName, wordColumnName)
+    this.distLdaModel = distLdaModel
+  }
+
+  /**
+   * Create Tk Lda Model
+   */
+  def this(distLdaModel: DistributedLDAModel,
```
Comment (joyeshmishra) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala]: - [ ] Testing markdowns

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala (hunk 3 above)
Comment (fmagnott) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala]: test

File: core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala [hunk 5]
```diff
@@ -0,0 +1,499 @@
...
+import org.trustedanalytics.sparktk.saveload.SaveLoad
+
+import scala.collection.immutable.Map
+import scala.util.Try
+
+/**
+ * Model for Latent Dirichlet Allocation. This holds the Spark's DistributedLdaModel and other artifacts obtained
+ * during model training
+ *
+ * @param numTopics Number of topics in trained model
```
Comment (fmagnott) [inline core/src/main/scala/org/apache/spark/mllib/clustering/org/trustedanalytics/sparktk/TkLdaModel.scala]: test

**Pull request merged**